tests/framework/results/test_context.py

```diff
@@ -196,14 +206,224 @@ def test_add_observation_nop_stratifications(default, additional, excluded, matc
+        ctx.add_stratification("house", ["house"], CATEGORIES, None, True)
+    if "familiar" in stratifications:
+        ctx.add_stratification("familiar", ["familiar"], FAMILIARS, None, True)
+    ctx.add_observation(
+        name,
+        pop_filter,
+        aggregator_sources,
+        aggregator,
+        stratifications,
+        [],
+        event_name,
+    )
+
+    i = 0
+    for _ in ctx.gather_results(population, event_name):
+        i += 1
+    assert i == 1
+
+
+@pytest.mark.parametrize(
+    "name, pop_filter, aggregator_sources, aggregator, stratifications",
+    [
+        ("wizard_count", "tracked==True", None, len, ["house", "familiar"]),
+        ("power_level_total", "tracked==True", ["power_level"], sum, ["house", "familiar"]),
+        (
+            "wizard_time",
+            "tracked==True",
+            [],
+            _aggregate_state_person_time_function,
+            ["house", "familiar"],
+        ),
+        ("wizard_count", "tracked==True", None, len, ["familiar"]),
+        ("power_level_total", "tracked==True", ["power_level"], sum, ["familiar"]),
+        (
+            "wizard_time",
+            "tracked==True",
+            [],
+            _aggregate_state_person_time_function,
+            ["familiar"],
+        ),
+    ],
+    ids=[
+        "len_aggregator_two_stratifications",
+        "sum_aggregator_two_stratifications",
+        "custom_aggregator_two_stratifications",
+        "len_aggregator_one_stratification",
+        "sum_aggregator_one_stratification",
+        "custom_aggregator_one_stratification",
+    ],
+)
+def test_gather_results_partial_stratifications_in_results(
+    name, pop_filter, aggregator_sources, aggregator, stratifications
+):
+    """Test cases where not all stratifications are observed for gather_results"""
+    ctx = ResultsContext()
+
+    # Generate population DataFrame
+    population = BASE_POPULATION.copy()
+
+    # Mock out some extra columns that would be produced by the manager's _prepare_population() method
+    population["current_time"] = pd.Timestamp(year=2045, month=1, day=1, hour=12)
+    population["step_size"] = timedelta(days=28)
+    population["event_time"] = pd.Timestamp(year=2045, month=1, day=1, hour=12) + timedelta(
+        days=28
+    )
+    # Remove an entire category from a stratification
+    population = population[population["familiar"] != "unladen_swallow"].reset_index()
+
+    event_name = "collect_metrics"
+
+    # Set up stratifications
+    if "house" in stratifications:
+        ctx.add_stratification("house", ["house"], CATEGORIES, None, True)
+    if "familiar" in stratifications:
+        ctx.add_stratification("familiar", ["familiar"], FAMILIARS, None, True)
+    ctx.add_observation(
+        name,
+        pop_filter,
+        aggregator_sources,
+        aggregator,
+        stratifications,
+        [],
+        event_name,
+    )
+
+    for r in ctx.gather_results(population, event_name):
+        assert (
```

Comment:
Correct me if I'm wrong, but this is testing that the stratifications containing unladen swallow are present in the results here? Shouldn't we also assert that they are 0?

Reply:
Correct. We could assert that, probably, but I hesitated because assumption of the `aggregator`.